lib/BC/Util.cpp

```diff
@@ -2437,6 +2433,7 @@ llvm::Value *BuildPointerToOffset(llvm::IRBuilder<> &ir, llvm::Value *ptr,
     }
   }
 
+  // NOTE(alex): Same here
   const auto dest_elem_type = PointerElementType(dest_elem_ptr_type);
```

Comment:
One problem area is here: https://github.com/lifting-bits/anvill/blob/master/lib/Lifters/DataLifter.cpp#L112
If we want to pass the underlying element types through to this function, we're going to need them here. However, one of the pointer types comes from a `VariableDecl` which I believe actually gets parsed from the JSON format. So we can't really embed the element type into the `VariableDecl` because we purposely removed that information from the JSON spec format.

Reply:
I'd like to say that the anvill side of things should stop doing that, whatever "that is." And by this, I mean: I think we should stop lifting data and global vars in anvill, and try make that function return `nullptr`, or pass in a `nullptr` type or something.

But, that is beyond the scope of this change, so I would err on the side of: maybe the subsequent code should stop trying to be clever, and instead just do byte-granularity indexing, instead of pointer element type size-granularity indexing.

Reply:
I'm interpreting that to mean that I should change `BuildPointerToOffset` index at the byte-level instead of doing clever things with the element size as opposed to writing something custom to replace that invocation of `BuildPointerToOffset`.

I have some code to do this and am currently debugging it.

Reply:
[080969e](https://github.com/lifting-bits/remill/pull/596/commits/080969e0b00fe3ea80d04792b26d668bf602a352)